### PR TITLE
Implement php parser

### PR DESCRIPTION
**Pull request by CDluznie**
Implementation of the PHP parser

### CODE REVIEW

File: tree_hugger/core/parser/php/php_parser.py [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
+    QUERY_FILE_PATH = Path(__file__).parent / "queries.yml"
 
     def __init__(self, library_loc: str=None, query_file_path: str=None):
-        query_file_content = pkg_resources.resource_string(__name__, PHPParser.QUERY_FILE_PATH)\
-                                          .decode("utf-8")
-        query = Query.fromString(query_file_content)
-        super(PHPParser, self).__init__('python', 'python_queries', query, library_loc)
+        super(PHPParser, self).__init__('php', 'php_queries', PHPParser.QUERY_FILE_PATH, library_loc)
+
+    def get_all_function_names(self) -> List:
```
Comment (rcshubhadeep) [inline tree_hugger/core/parser/php/php_parser.py:18]: `-> List[str]`

File: tree_hugger/core/parser/php/php_parser.py [hunk 2]
```diff
@@ -37,17 +44,83 @@ def get_all_class_method_names(self) -> List:
...
-        all_methods = set([method_name  for key, value in methods.items() for method_name in value])
+        Excludes any methods, i.e., functions defined inside a class.
+        """
+        function_names = self.get_all_function_names()
+        func_and_params = self.function_names_with_params()
+        
+        captures = self._run_query_and_get_captures('all_function_bodies', self.root_node)
+        
+        ret_struct = {}
+        pp = {}
```
Comment (rcshubhadeep) [inline tree_hugger/core/parser/php/php_parser.py:66]: I had probably wrote this variable name. Not the best one. If you wish you can change it.

File: tree_hugger/core/parser/php/php_parser.py [hunk 3]
```diff
@@ -37,17 +44,83 @@ def get_all_class_method_names(self) -> List:
...
+
+        for i in range(0, len(captures), 2):
+            func_name = match_from_span(captures[i][0], self.splitted_code)
+            if func_name in function_names:
+                params = match_from_span(captures[i+1][0], self.splitted_code)
+                ret_struct[func_name] = params
+        
+        return ret_struct
+
+    def _walk_recursive_documentation(self, cursor: TreeCursor, lines: List, node_type, documented: Dict):
```
Comment (rcshubhadeep) [inline tree_hugger/core/parser/php/php_parser.py:90]: `node_type: str` maybe

**Comment by CDluznie** [comment]
Probably we also need to rename the function inside the file `queries.yml` (docstring to documentation)

**Pull request merged**